core/src/toga/screen.py

```diff
@@ -0,0 +1,18 @@
+class Screen:
+    _screen_interface_cache = {}
+
+    def __new__(cls, _impl):
+        if _impl in cls._screen_interface_cache:
+            return cls._screen_interface_cache[_impl]
+        else:
+            instance = super().__new__(cls)
+            cls._screen_interface_cache[_impl] = instance
+            return instance
+
+    def __init__(self, _impl):
+        if not hasattr(self, "_impl"):
+            self._impl = _impl
```

Comment:
Same comment as above about the need for `__init__`; plus, if we're constructing an new interface object wrapper around an impl, we also know that the impl can have a reference to the interface.

Reply:
Done